kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaProxyContext.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.time.Clock;
+import java.util.stream.Collectors;
+
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.VirtualKafkaCluster;
+import io.kroxylicious.kubernetes.operator.model.ProxyModel;
+import io.kroxylicious.kubernetes.operator.model.ProxyModelBuilder;
+
+public class KafkaProxyContext {
+
+    private static final String KEY_SEC = "sec";
+    private static final String KEY_MODEL = "model";
+    private static final String KEY_CLOCK = "clock";
+
+    private KafkaProxyContext() {
+    }
+
+    static void init(Clock clock,
+                     SecureConfigInterpolator secureConfigInterpolator,
+                     KafkaProxy proxy,
+                     Context<KafkaProxy> context) {
+        var rc = context.managedWorkflowAndDependentResourceContext();
+
+        rc.put(KEY_CLOCK, Clock.fixed(clock.instant(), clock.getZone()));
+
+        rc.put(KEY_SEC, secureConfigInterpolator);
+
+        ProxyModelBuilder proxyModelBuilder = ProxyModelBuilder.contextBuilder();
+        ProxyModel model = proxyModelBuilder.build(proxy, context);
+        rc.put(KEY_MODEL, model);
+    }
+
+    static Clock clock(Context<KafkaProxy> context) {
+        return context.managedWorkflowAndDependentResourceContext().getMandatory(KEY_CLOCK, Clock.class);
+    }
+
+    static SecureConfigInterpolator secureConfigInterpolator(Context<KafkaProxy> context) {
+        return context.managedWorkflowAndDependentResourceContext().getMandatory(KEY_SEC, SecureConfigInterpolator.class);
+    }
+
+    static ProxyModel model(Context<KafkaProxy> context) {
+        return context.managedWorkflowAndDependentResourceContext().getMandatory(KEY_MODEL, ProxyModel.class);
+    }
+
+    static boolean isBroken(Context<KafkaProxy> context, VirtualKafkaCluster cluster) {
+        var model = KafkaProxyContext.model(context);
+        var fullyResolved = model.resolutionResult().fullyResolvedClustersInNameOrder().stream().map(ResourcesUtil::name).collect(Collectors.toSet());
+        return !(fullyResolved.contains(ResourcesUtil.name(cluster))
```

Comment:
so it's broken if the cluster isn't fully resolved && all ingresses for cluster had no errors (like conflicts)? Feels wrong, maybe I need more coffee. Isn't a VKC always broken if it's not fully resolved?

Maybe the logic should be the VKC is broken if `!fullyResolved || allIngressesHaveExceptions` ?

Reply:
> Do you really mean it's broken only if all ingresses have exceptions? Or should the cluster be considered broken if any have exceptions?

It depends, maybe we could stand up a VKC that has ingress A working, despite ingress B having some exception. Lets start restrictive and require all ingresses to have no errors.

> why did you make the signature Set<IngressConflictException> ingressExceptions() on ProxyIngressModel?

I was trying to make `ProxyIngressModel` describe everything that went wrong during a model build and leave it up to the client/reconciler to figure out what it means. But happy to have conveniences like `Optional<ICE> firstException()` instead.